frontends/p4/fromv1.0/programStructure.h

```diff
@@ -169,7 +169,35 @@ class ProgramStructure {
  protected:
     const IR::Statement* convertPrimitive(const IR::Primitive* primitive);
     void checkHeaderType(const IR::Type_StructLike* hrd, bool toStruct);
-    const IR::Annotations* addNameAnnotation(cstring name, const IR::Annotations* annos = nullptr);
+
+    /**
+     * Extend the provided set of annotations with an '@name' annotation for the
+     * given name. This is used to preserve the original, P4-14 object names.
+     *
+     * In general, because P4-14 names do not have the hierarchical structure
```

Comment:
The P4-14 names are not hierarchical, with one exception which is not specified in the P4-14 spec: calling a control multiple times. It is not clear what that means, and how names should be assigned for the tables inside the called control.